gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java

```diff
@@ -726,6 +741,230 @@ private final void deleteTopic(DeleteTopicRequest request) {
     return deleteTopicCallable;
   }
 
+  // AUTO-GENERATED DOCUMENTATION AND METHOD
+  /**
+   * Sets the access control policy on the specified resource. Replaces any
+   * existing policy.
+   *
+   * Sample code:
+   * <pre><code>
+   * try (PublisherApi publisherApi = PublisherApi.create()) {
+   *   String formattedResource = PublisherApi.formatTopicName("[PROJECT]", "[TOPIC]");
+   *   Policy policy = Policy.newBuilder().build();
+   *   Policy response = publisherApi.setIamPolicy(formattedResource, policy);
+   * }
+   * </code></pre>
+   *
+   * @param resource REQUIRED: The resource for which policy is being specified.
+   * Resource is usually specified as a path, such as,
+   * projects/{project}/zones/{zone}/disks/{disk}.
+   * @param policy REQUIRED: The complete policy to be applied to the 'resource'. The size of
+   * the policy is limited to a few 10s of KB. An empty policy is in general a
+   * valid policy but certain services (like Projects) might reject them.
+   * @throws com.google.api.gax.grpc.ApiException if the remote call fails
+   */
+  public final Policy setIamPolicy(String resource, Policy policy) {
```

Comment:
Documentation here (and for other methods) is very vague IMHO. Also I believe that the parameter name `resource` should be called `topic`. Also the example for the `resource` parameter (`projects/{project}/zones/{zone}/disks/{disk}`) has nothing to do with Pub/Sub or topics and might be misleading.
I see that also [the official docs](https://cloud.google.com/pubsub/reference/rest/v1/projects.subscriptions/setIamPolicy) are very vague, which makes me think that your hands are tied by auto-generation. It might be useful to raise this issue to the people writing the docs to make them more specific.